src/coreclr/src/interop/inc/interoplibimports.h

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _INTEROP_INC_INTEROPLIBIMPORTS_H_
+#define _INTEROP_INC_INTEROPLIBIMPORTS_H_
+
+#include "interoplib.h"
+
+namespace InteropLibImports
+{
+    enum class AllocScenario
+    {
+        ManagedObjectWrapper,
+        NativeObjectWrapper,
+    };
+
+    // Allocate the given amount of memory.
+    void* MemAlloc(_In_ size_t sizeInBytes, _In_ AllocScenario scenario) noexcept;
+
+    // Free the previously allocated memory.
+    void MemFree(_In_ void* mem, _In_ AllocScenario scenario) noexcept;
+
+    // Add memory pressure to the runtime's GC calculations.
+    HRESULT AddMemoryPressureForExternal(_In_ UINT64 memoryInBytes) noexcept;
+
+    // Remove memory pressure from the runtime's GC calculations.
+    HRESULT RemoveMemoryPressureForExternal(_In_ UINT64 memoryInBytes) noexcept;
+
+    enum class GcRequest
+    {
+        Default,
+        Extensive // This is an expensive GC request, akin to a Gen2/"stop the world" GC.
```

Comment:
"extensive GC" is not a term I have ever heard in connection with GC.

This does full blocking GC. Should this be called `FullBlocking`?

Reply:
Yep. I was struggling with a name. I can do `FullBlocking`. Will see if this is all green first and then change it.